components/Note/index.js

```diff
@@ -0,0 +1,105 @@
+import styled from "styled-components";
+import useSWR from "swr";
+import { useState } from "react";
+import NotesForm from "../NotesForm";
+import { useRouter } from "next/router";
+
+const StyledListItem = styled.li`
+  background-color: rgba(255, 255, 255, 0.6);
+  padding: 10px;
+  display: flex;
+  flex-direction: column;
+  align-content: center;
+`;
+
+export default function Note({ note, locatData }) {
+  const router = useRouter();
+  const { mutate } = useSWR(`/api/locations/${locatData?._id}`);
+  const [isEditMode, setIsEditMode] = useState(false);
+
+  async function handleEditNote(event) {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const noteData = Object.fromEntries(formData);
+
+    const response = await fetch(`/api/notes/${note?._id}`, {
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(noteData),
+    });
+    if (response.ok) {
+      mutate();
+      setIsEditMode(false);
+    }
+  }
+  async function handleDeleteNote() {
+    const responseNote = await fetch(`/api/notes/${note?._id}`, {
+      method: "DELETE",
+    });
+    if (!responseNote.ok) {
+      return <h1>Something gone wrong!</h1>;
+    }
+
+    if (responseNote.ok) {
+      const responseLocation = await fetch(`/api/locations/${locatData?._id}`, {
+        method: "PATCH",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          notes: locatData?.notes.filter((oneNote) => {
+            return oneNote._id == note._id ? false : true;
+          }),
+        }),
+      });
+      if (responseLocation.ok) {
+        mutate();
```

Comment:
You don't need to mutate before routing since routing automatically sends requests.

Reply:
done